Cookbook/Proposals/ComposableReducers.md

```diff
@@ -0,0 +1,250 @@
+
+```swift
+struct Feedback<Action, State> {
+    let transform: (Signal<State>) -> Signal<Action>
+
+    /// Zero value for `+`.
+    static var empty: Feedback {
+        return Feedback { _ in .empty }
+    }
+
+    // Append operator, just like `+`.
+    static func <> (lhs: Feedback, rhs: Feedback) -> Feedback {
+        return Feedback { state in
+            Signal.merge(lhs.transform(state), rhs.transform(state))
+        }
+    }
+}
+
+extension Feedback {
+    public func lift<SuperState>(state lens: Lens<SuperState, State>) -> Feedback<Action, SuperState> {
+        return Feedback<Action, SuperState> { superState in
+            self.transform(superState.map(lens.get))
+        }
+    }
+
+    public func lift<SuperAction>(action prism: Prism<SuperAction, Action>) -> Feedback<SuperAction, State> {
+        return Feedback<SuperAction, State> { state in
+            self.transform(state).map(prism.review)
+        }
+    }
+}
+```
+
+## Example
+
+```swift
+// MARK: - Component 1 (isolated from Main & Component 2)
+//---------------------------------------------------------
+
+enum Sub1Action {
+    case increment
+    case decrement
+}
+
+struct Sub1State {
+    var count: Int = 0
+}
+
+let subReducer1 = Reducer<Sub1Action, Sub1State> { action, state in
+    switch action {
+    case .increment: return state.with { $0.count + 1 }
+    case .decrement: return state.with { $0.count - 1 }
+    }
+}
+
+let subFeedback1: Feedback<Sub1Action, Sub1State> = .empty
+
+// MARK: - Component 2 (isolated from Main & Component 1)
+//---------------------------------------------------------
+
+enum Sub2Action { ... }
+struct Sub2State { ... }
+let subReducer2: Reducer<Sub2Action, Sub2State> = ...
+let subFeedback2: Feedback<Sub2Action, Sub2State> = ...
+
+// MARK: - Main
+//---------------------------------------------------------
+
+enum MainAction {
+    case sub1(Sub1Action)
+    case sub2(Sub2Action)
+    ...
+}
+
+extension Prism where Whole == MainAction, Part == Sub1Action {
+    static let sub1Action = Prism(
+        preview: {
+            guard case let .sub1Action(action) = $0 else { return nil }
+            return action
+        },
+        review: MainAction.sub1Action
+    )
+}
+
+...
+
+struct MainState {
+    var sub1: Sub1State
+    var sub2: Sub2State
+    // var shared: ...  /* NOTE: Shared state can belong to here */
+}
+
+extension Lens where Whole == MainState, Part == Sub1State {
+    static let sub1State = Lens(
+        get: { $0.sub1 },
+        set: { whole, part in
+            whole.with { $0.sub1 = part }
+        }
+    )
+}
+
+...
+
+let mainReducer: Reducer<MainAction, MainState> =
+    subReducer1
+        .lift(action: .sub1Action)
+        .lift(state: .sub1State)
+    <>
+    subReducer2
+        .lift(action: .sub2Action)
+        .lift(state: .sub2State)
+
+let mainFeedback: Feedback<MainAction, MainState> =
```

Comment:
maybe it will be more familiar to everyone if we compose feedbacks via array literals?

```swift
let mainFeedback: Feedback<MainAction, MainState> = [
  subFeedback1
    .lift(action: .sub1Action)
    .lift(state: .sub1State),
  subFeedback2
    .lift(action: .sub2Action)
    .lift(state: .sub2State)
]
```

Reply:
And we could use variating argument in the State property constructor to avoid extra `[]`

```
state = Property(
            initial: State(
                prescriptionID: prescriptionID,
                status: initialStatus
            ),
            scheduler: scheduler,
            reduce: PrescriptionViewModel.reduce,
            feedbacks: 
	        	    subFeedback1
	        	        .lift(action: .sub1Action)
	        	        .lift(state: .sub1State),
	        	    subFeedback2
	        	        .lift(action: .sub2Action)
	        	        .lift(state: .sub2State)
        )
```

Reply:
@ilyapuchka 

`ExpressibleByArrayLiteral` can be harmful when we want to really create `Array<Reducer>` but it forces us to concatenates into a single `Reducer`.
(Or Swift compiler might claim for ambiguous usage)

And variadic arguments is about improvement for `ReactiveFeedback` framework and not about this proposal.

Reply:
> And variadic arguments is about improvement for ReactiveFeedback framework and not about this proposal.

we are doing changes to the Feedback type already with this proposal (either through extension or in the ReactiveFeedback directly) so we could add such extension as well 🤷‍♂ (though in fact Swift only supports one variadic argument, so it won't work). I think the simpler we make this to use for everyone the better, but I guess this can be discussed if we get to the actual implementation.